nutanix/services/dataprotectionv2/resource_nutanix_promote_protected_resource_v2_test.go

```diff
@@ -0,0 +1,465 @@
+  availability_zone = local.config.availability_zone
+}
+
+# Create Category
+resource "nutanix_category_v2" "test" {
+  key   = "tf-synchronous-pp-%[5]d"
+  value = "tf_synchronous_pp_%[5]d"
+}
+
+resource "nutanix_protection_policy_v2" "test" {
+  name        = "%[4]s"
+  description = "%[3]s"
+
+  replication_configurations {
+    source_location_label = "source"
+    remote_location_label = "target"
+    schedule {
+      recovery_point_type                           = "CRASH_CONSISTENT"
+      recovery_point_objective_time_seconds         = 0
+      sync_replication_auto_suspend_timeout_seconds = 10
+    }
+  }
+  replication_configurations {
+    source_location_label = "target"
+    remote_location_label = "source"
+    schedule {
+      recovery_point_type                           = "CRASH_CONSISTENT"
+      recovery_point_objective_time_seconds         = 0
+      sync_replication_auto_suspend_timeout_seconds = 10
+    }
+  }
+
+  replication_locations {
+    domain_manager_ext_id = data.nutanix_pcs_v2.pcs-list.pcs[0].ext_id
+    label                 = "source"
+    is_primary            = true
+    replication_sub_location {
+      cluster_ext_ids {
+        cluster_ext_ids = [local.clusterExtId]
+      }
+    }
+  }
+  replication_locations {
+    domain_manager_ext_id = local.availability_zone.pc_ext_id
+    label                 = "target"
+    is_primary            = false
+    replication_sub_location {
+      cluster_ext_ids {
+        cluster_ext_ids = [local.availability_zone.cluster_ext_id]
+      }
+    }
+  }
+
+  category_ids = [nutanix_category_v2.test.id]
+}
+
+resource "nutanix_virtual_machine_v2" "test" {
+  name                 = "%[2]s"
+  description          = "%[3]s"
+  num_cores_per_socket = 1
+  num_sockets          = 1
+  cluster {
+    ext_id = data.nutanix_clusters_v2.clusters.cluster_entities.0.ext_id
+  }
+  categories {
+    ext_id = nutanix_category_v2.test.id
+  }
+  power_state = "OFF"
+  depends_on = [nutanix_protection_policy_v2.test]
+}
+
+
+	`, filepath, vmName, description, ppName, r)
+}
+
+func testPromoteProtectedResourceVMConfig() string {
+	remotePcIP := testVars.AvailabilityZone.RemotePcIP
+
+	username := os.Getenv("NUTANIX_USERNAME")
+	password := os.Getenv("NUTANIX_PASSWORD")
+	port, _ := strconv.Atoi(os.Getenv("NUTANIX_PORT"))
+	insecure, _ := strconv.ParseBool(os.Getenv("NUTANIX_INSECURE"))
+	remoteHostProviderConfig := fmt.Sprintf(`
+provider "nutanix-2" {
+  username = "%[1]s"
+  password = "%[2]s"
+  endpoint = "%[3]s"
+  insecure = %[4]t
+  port     = %[5]d
+}
+
+`, username, password, remotePcIP, insecure, port)
+
+	return fmt.Sprintf(
+		`
+
+
+%[1]s
+
+resource "nutanix_promote_protected_resource_v2" "test" {
+  provider = nutanix-2
+  ext_id = nutanix_virtual_machine_v2.test.id
+}
+
+`, remoteHostProviderConfig)
+}
+
+func testPromoteProtectedResourceVGAndProtectionPolicyConfig(clusterName, vgName, ppName, description, categoryKey, categoryValue string) string {
```

Comment:
Please document what are the steps done in test script so that reader can understand the workflow. You can have summary above the function name. Also mention what are test requirements.